sdv/constraints/tabular.py

```diff
@@ -531,6 +541,11 @@ def reverse_transform(self, table_data):
             original_column = self._value - diff_column
 
         table_data[self._column_name] = pd.Series(original_column).astype(self._dtype)
+        if self._is_datetime and self._datetime_format:
```

Comment:
Sometimes `self._datetime_format` may not be learnt (it can return `None`) since the learning of the format does not support `ms`: 
```
In [32]: detect_datetime_format('2021-02-02T00:00:00.000000')
In [33]: detect_datetime_format('2021-02-02T00:00:00')
Out[33]: '%Y-%m-%dT%H:%M:%S'
```